### PR TITLE
Kaline tree level up

### LINKED ISSUE #374: [FEATURE] Kaline Tree level up
## 🚀 Feature

Level up the Kaline Tree, spending gold and fertilizer when doing so.
In the future, leveling up will unlock game modes and features.

### PR DESCRIPTION
**Pull request by ncontinanza**
Closes #374 

⚠️ Test using this branch on the backend: https://github.com/lambdaclass/mirra_backend/pull/499

## Summary of changes

Implement the level up of the Kaline Tree, as well as the display of the leveling costs in the Kaline Tree scene.

## How has this been tested?

1. Start the game and go to the Kaline Tree.
2. Initially, you have enough Fertilizer and Gold to level up the tree once, so you should be able to do it.
3. After leveling up, check that gold and fertilizer were consumed by running `GameBackend.Users.Currencies.get_user_currency(user_id, GameBackend.Users.Currencies.get_currency_by_name!("Gold").id)` and `GameBackend.Users.Currencies.get_user_currency(user_id, GameBackend.Users.Currencies.get_currency_by_name!("Fertilizer").id)` with the respective user_id.
4. Make sure that the gold in the header was updated, and that the leveling costs increased. As we haven't decided a formula for upgrading yet, leveling costs are equal to `100 * kaline_tree_level`.
5. Try to level up once more and verify that there's an error shown in a pop up.

## Checklist
- [X] I have tested the changes locally.
- [X] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

**Review by martinacantaro — APPROVED** [review]
Functional review approved 👍 
For the next PR, please add the Tree level in the frontend

**Review by lotuuu — CHANGES_REQUESTED** [review]
There's a bug on the level up cost. When I'm not able to level up any more, if I get out of the scene and back in, the level up cost is reset to the amount it displayed when I go into it.


https://github.com/lambdaclass/champions_of_mirra/assets/31993243/0815c10f-3a19-4343-a648-d194814fba13

File: gateway.proto [hunk 1]
```diff
@@ -164,6 +169,15 @@ option csharp_namespace = "Protobuf.Messages";
     repeated Unit units = 8;
     repeated Item items = 9;
     repeated AfkRewardRate afk_reward_rates = 10;
+    KalineTreeLevel kaline_tree_level = 11;
+  }
+
+  message KalineTreeLevel {
+    string id = 1;
+    uint64 level = 2;
+    uint64 fertilizer_level_up_cost = 3;
```
[the remark this answers is not in the document]
Comment (ncontinanza) [reply gateway.proto:178]: As far as I checked, we do not. Probably because we're not receiving that info from the backend, I'll create an issue for that

**Review by lotuuu — DISMISSED** [review]

**Review by martinacantaro — DISMISSED** [review]
Re-tested via frontend and backend commands. Functional review approved!

**Review by tkz00 — CHANGES_REQUESTED** [review]
Great, works amazing. Just one thing, fields should be named using camelCase, not snake_case, according to the project's and C#'s conventions.

**Review by tkz00 — APPROVED** [review]
Amazing!

**Review by martinacantaro — APPROVED** [review]
approved once again

**Pull request merged**